Fix hanging api call, parsing errors table bug, and update tests

cwcrystal8 opened the pull request:
1. Fixed a bug where the parsing errors table would appear even though there was no error. The problem with the code was it was missing a check for errors before adding them to the list.

2. Fixed a bug where the node page would fail to render properly. The problem was a missing check for non-null labels before iterating through them.
Switch over to new API

3. Updated the test cases and source code for changes in the behavior of the API. 

4. Reran build and updated static files to reflect the new src changes.

pradh commented on mcf-browser/src/back-end/server-api.js:
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
+        if(tmcfOut['errMsgs'] && tmcfOut['errMsgs'].length !== 0) {
+
+          finalReturn['errMsgs'] =
           finalReturn['errMsgs'].concat({
```
The indents seem off, could you run lint?

cwcrystal8 replied:
Is there a specific linter we use in DC?

pradh replied:
Website uses this - https://github.com/datacommonsorg/website/blob/master/run_test.sh#L49-L52

pradh commented on mcf-browser/src/back-end/utils.js:
```diff
@@ -106,7 +106,7 @@ async function doesExistsInKG(dcid) {
   return fetch(url)
       .then((res) => res.json())
-      .then((data) => JSON.parse(data.payload)[dcid] ? true : false);
+      .then((data) => (JSON.parse(data.payload)[dcid] && JSON.parse(data.payload)[dcid].length > 0) ? true : false);
```
Perhaps this could use a comment?

cwcrystal8 replied:
Added one!

pradh commented on mcf-browser/src/back-end/server-api.js:
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
           });
-        finalReturn['errMsgs'] =
+
+        if(tmcfOut['errMsgs'] && tmcfOut['errMsgs'].length !== 0) {
```
Just wondering why just having the second condition may not work?  Given https://github.com/datacommonsorg/tools/blob/4bd98c15a4fd309ece98301da4946234da24b61a/mcf-browser/src/back-end/parse-mcf.js#L69

pradh reviewed (commented):
Nice first PR!

Does this PR actually include the switch to new API as the description suggests or did I miss it?

cwcrystal8 commented:
ptal, the code has the following changes:

- Fixed indentation
- Removed unnecessary condition in parsing error if statement
- Added a comment explaining the logic behind checking for DCIDs

Also, @pradh this PR does not include the switch to the new API, it just fixes the back-end issue in server-api.js. New API switch will be in the next PR, sorry for the confusion!

pradh commented:
> 

Ok, could update the description then: https://github.com/datacommonsorg/website/blob/master/run_test.sh#L49-L52

pradh reviewed (dismissed):
Thanks for the changes!

cwcrystal8 commented:
Updated the description! I just realized you were referring to point (3), which was confusingly worded. Sometime between the last update and this PR, the behavior of the current (old) API changed so I updated the test cases to reflect that new behavior. I reworded point (3) to make this clearer.

cwcrystal8 commented on mcf-browser/asset-manifest.json:
```diff
@@ -1,22 +1,23 @@
 {
```
Yep, the file was automatically generated using `npm run build`

cwcrystal8 commented on mcf-browser/src/back-end/server-api.js:
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
-          });
+
+        if(tmcfOut['errMsgs'].length !== 0) {
+
```
fixed in next commit

cwcrystal8 commented on mcf-browser/src/back-end/server-api.js:
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
-            'errs': tmcfOut['errMsgs'],
-          });
+
+        if(tmcfOut['errMsgs'].length !== 0) {
```
fixed in next commit

cwcrystal8 commented on mcf-browser/src/back-end/graph.js:
```diff
@@ -178,6 +178,10 @@ class Node {
    *     the calling Node is the source of the triple.
    */
   async createAssertionsFromLabels(propLabels, isInverse) {
+    if(!propLabels || propLabels.length === 0){
```
fixed in next commit

cwcrystal8 commented:
@beets ptal, the whitespace issues should be fixed now
